Fix heatmap

kvpt opened the pull request:
The date interval generation was wrong for loading the heatmap.
The UTC date was not correctly generated.
The lastDayOfMonth was given only the month and not the complete date.

### Before
![Heatmap_Before](https://user-images.githubusercontent.com/1446221/82122499-5da0e180-9794-11ea-862b-455237a92667.PNG)

### After
![Heatmap_After](https://user-images.githubusercontent.com/1446221/82122503-609bd200-9794-11ea-885f-c48c8a6a3252.PNG)

DanielRuf commented on frontend/src/components/Service/ServiceHeatmap.vue:
```diff
@@ -82,18 +82,16 @@
       methods: {
           async chartHeatmap() {
-              let start = new Date(new Date().getUTCFullYear(), new Date().getUTCMonth()-2, 1);
+              let start = new Date(Date.UTC(new Date().getUTCFullYear(), new Date().getUTCMonth()-2, 1));
```
I think this can be simplified.

kvpt replied:
Without the Date.UTC the date is not correctly initialized.
![image](https://user-images.githubusercontent.com/1446221/82125637-841c4800-97a7-11ea-9562-a7486ff64fb4.png)

![image](https://user-images.githubusercontent.com/1446221/82125662-af069c00-97a7-11ea-93ef-1265db6d937a.png)

DanielRuf replied:
With simplified I mean less duplicated code parts and a few more variables ;-)

DanielRuf commented on frontend/src/components/Service/ServiceHeatmap.vue:
```diff
@@ -82,18 +82,16 @@
                   const inputdata = await this.heatmapData(start, end)
                   monthData.push(inputdata)
-                  start = new Date(start.getUTCFullYear(), start.getUTCMonth()+1, 1);
+                  start = new Date(Date.UTC(start.getUTCFullYear(), start.getUTCMonth()+1, 1));
```
As the get... methods are already called earlier we can also use variables for these instead of recalculating the values.

kvpt replied:
I thought of using another approach, but I preferred to stay as close as possible to the current code.
The code related to date are always error prone and hard to get right.

hunterlong commented:
@kvpt , thank you for this PR, **I seriously appreciate it**. Feel free to change what you feel is needed, creating duplicate code is something I believe we both don't want. And ya... I'm so over dealing with time within Browser->Frontend->Backend->Database then back again. Basically, just looking at time is a migraine. haha

I seem to be putting time based functionality within the [frontend/muxin.js](https://github.com/statping/statping/blob/master/frontend/src/mixin.js) file. We are also using [date-fns](https://date-fns.org/) to calculate time on the frontend.

kvpt commented:
I refactored the code to be more simple and delegate the date calculation to the date-fns library.
I also removed the UTC part because the user want to have the data relative to his timezone.
